framework/src/mesh/PatternedMesh.C

```diff
@@ -0,0 +1,164 @@
+
+  _meshes.resize(_files.size());
+
+  // Read in all of the meshes
+  for (unsigned int i = 0; i < _files.size(); i++)
+  {
+    SerialMesh * mesh = new SerialMesh(_communicator);
+    mesh->read(_files[i]);
+
+    _meshes[i] = mesh;
+  }
+
+  _row_meshes.resize(_pattern.size());
+
+  // The zeroth row (which is the top row) will be the real mesh
+  _row_meshes[0] = dynamic_cast<SerialMesh*>( &getMesh() );
+
+  // Create a mesh for all the other rows
+  for (unsigned int i = 1; i < _pattern.size(); i++)
+  {
+    SerialMesh * mesh = new SerialMesh(_communicator);
+    _row_meshes[i] = mesh;
+  }
+}
+
+PatternedMesh::PatternedMesh(const PatternedMesh & other_mesh) :
+    MooseMesh(other_mesh),
+    _files(other_mesh._files),
+    _pattern(other_mesh._pattern),
+    _x_width(other_mesh._x_width),
+    _y_width(other_mesh._y_width),
+    _z_width(other_mesh._z_width)
+{
+}
+
+PatternedMesh::~PatternedMesh()
+{
+  // Clean up the mesh we made (see what I did there?)
+  for (unsigned int i = 0; i < _meshes.size(); i++)
+    delete _meshes[i];
+
+  // Don't delete the first row - it is the real mesh - it will be cleaned up later
+  for (unsigned int i = 1; i < _meshes.size(); i++)
+    delete _row_meshes[i];
+}
+
+
+MooseMesh &
+PatternedMesh::clone() const
+{
+  return *(new PatternedMesh(*this));
+}
+
+void
+PatternedMesh::buildMesh()
+{
+  // stitch_meshes() is only implemented for SerialMesh.  So make sure
+  // we have one here before continuing.
+  SerialMesh* the_mesh = dynamic_cast<SerialMesh*>( &getMesh() );
+
+  if (!the_mesh)
+    mooseError("Error, PatternedMesh calls stitch_meshes() which only works on SerialMesh.");
+  else
+  {
+    BoundaryID left = getBoundaryID(getParam<BoundaryName>("left_boundary"));
+    BoundaryID right = getBoundaryID(getParam<BoundaryName>("right_boundary"));
+    BoundaryID top = getBoundaryID(getParam<BoundaryName>("top_boundary"));
+    BoundaryID bottom = getBoundaryID(getParam<BoundaryName>("bottom_boundary"));
+    BoundaryID front = getBoundaryID(getParam<BoundaryName>("front_boundary"));
+    BoundaryID back = getBoundaryID(getParam<BoundaryName>("back_boundary"));
+
+    // Build each row mesh
+    for (int i = 0; i < _pattern.size(); i++)
+    {
+      for (int j = 0; j < _pattern[i].size(); j++)
+      {
+        // If this is the first cell of the row initialize the row mesh
+        if (j == 0)
+        {
+          _row_meshes[i]->read(_files[_pattern[i][j]]);
+
+          MeshTools::Modification::translate(*_row_meshes[i], j*_x_width, -i*_y_width, 0);
+
+          continue;
+        }
+
+        SerialMesh & cell_mesh = *_meshes[_pattern[i][j]];
+
+        // Move the mesh into the right spot.  -i because we are starting at the top
+        MeshTools::Modification::translate(cell_mesh, j*_x_width, -i*_y_width, 0);
+
+        _row_meshes[i]->stitch_meshes(dynamic_cast<SerialMesh &>(cell_mesh), right, left, TOLERANCE, /*clear_stitched_boundary_ids=*/true);
```

Comment:
When stitching meshes, do we need to make sure the surface to be stitched has exactly the same nodes on both sides? Or will new elements will be created to avoid hanging nodes?